src/main/java/org/mockito/internal/creation/bytebuddy/InlineDelegateByteBuddyMockMaker.java

```diff
@@ -108,103 +109,111 @@ class InlineDelegateByteBuddyMockMaker
     private static final Throwable INITIALIZATION_ERROR;
 
     static {
-        Instrumentation instrumentation;
-        Throwable initializationError = null;
-
-        // ByteBuddy internally may attempt to fork a subprocess. In Java 11 and Java 19, the Java
-        // process class observes the os.name system property to determine the OS and thus determine
-        // how to fork a new process. If the user is stubbing System properties, they may clear
-        // the existing System properties, which will cause this to fail. This is very much an
-        // implementation detail, but it will result in Mockito failing to load with an error that
-        // is not overly clear, so let's attempt to detect this issue ahead of time instead.
-        if (System.getProperty("os.name") == null) {
-            throw new IllegalStateException(
-                    join(
-                            "The Byte Buddy agent cannot be loaded.",
-                            "",
-                            "To initialise the Byte Buddy agent, a subprocess may need to be created. To do this, the JVM requires "
-                                    + "knowledge of the 'os.name' System property in most JRE implementations. This property is not present, "
-                                    + "which means this operation will fail to complete. Please first make sure you are not clearing this "
-                                    + "property anywhere, and failing that, raise a bug with your JVM vendor."));
-        }
+        Instrumentation instrumentation = PremainAttach.getInstrumentation();
+        if (instrumentation != null && instrumentation.isRetransformClassesSupported()) {
+            INSTRUMENTATION = instrumentation;
+            INITIALIZATION_ERROR = null;
+        } else {
```

Comment:
I wonder if we should print a message when JVM is >= 21, that mockito should be explicitly added as a Java agent in the JVM running the tests due to https://openjdk.org/jeps/451.

Reply:
Also, I wonder if we should add a message lie this, or at least document this in the wiki or README, e.g. on pure gradle (there's also the option of additional plugins) 

```kotlin
val moclitoAgent = configurations.create("mockitoAgent")
dependencies {
    mockitoAgent(libs.mockito)
}
tasks {
    test {
        jvmArgs("-javaagent:${mockitoAgent.asPath}")
    }
}
```

And the same for maven, I'm not sure about the syntax I don't use maven anymore

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-dependency-plugin</artifactId>
    <executions>
        <execution>
            <goals>
                <goal>properties</goal>
            </goals>
        </execution>
    </executions>
</plugin>
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <argLine>@{argLine} -javaagent:${org.mockito:mockito-core:jar}</argLine>
    </configuration>
</plugin>
```

There might be something else with the dependency plugin to do.

Reply:
Good point. I think that we should point this out and give a solution for both popular build tools.